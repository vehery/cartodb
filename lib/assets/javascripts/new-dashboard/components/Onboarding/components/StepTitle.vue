<template>
  <h2 class="title is-body u-mb--24 u-flex u-flex__align--center">
    <span class="icon icon--large">
      <slot name="icon"></slot>
    </span>

    {{title}}
  </h2>
</template>

<script>
export default {
  name: 'StepTitle',
  props: {
    title: String
  }
};
</script>

<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.icon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 18px;
  height: 18px;
  margin-right: 16px;
  margin-left: -40px;
}

.icon--large {
  width: 24px;
  height: 24px;
}
</style>
