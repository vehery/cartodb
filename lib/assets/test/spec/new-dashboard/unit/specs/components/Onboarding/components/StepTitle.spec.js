import { shallowMount } from '@vue/test-utils';
import StepTitle from 'new-dashboard/components/Onboarding/components/StepTitle';

describe('StepTitle.vue', () => {
  it('should render properly', () => {
    const stepTitleComponent = shallowMount(StepTitle, {
      propsData: {
        title: 'Fake step title'
      },
      slots: {
        icon: '<img src="/fake-icon-url"/>'
      }
    });

    expect(stepTitleComponent).toMatchSnapshot();
  });
});
