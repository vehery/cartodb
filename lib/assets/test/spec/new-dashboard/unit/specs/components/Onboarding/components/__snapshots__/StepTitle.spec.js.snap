// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepTitle.vue should render properly 1`] = `
<h2 class="title is-body u-mb--24 u-flex u-flex__align--center"><span class="icon icon--large"><img src="/fake-icon-url"></span>

  Fake step title
</h2>
`;
